spec/controllers/rfid_controller_spec.rb

```diff
@@ -39,6 +39,54 @@
         expected = { error: 'Temporary RFID already exists' }
         expect(actual).to eq(expected)
       end
+
+      it 'should not create rfid and should sign in RFID and sign out' do
+        pi_reader = create(:pi_reader)
+        attributes = create(:rfid, mac_address: pi_reader.pi_mac_address)
+        rfid_params = { rfid: attributes.card_number, mac_address: attributes.mac_address }
+        expect { post :card_number, params: rfid_params, :format => :json }.to change(Rfid, :count).by(0)
+        actual = JSON.parse(response.body, symbolize_names: true)
+        expected = { success: 'RFID sign in' }
+        expect(actual).to eq(expected)
+        rfid_params = { rfid: attributes.card_number, mac_address: attributes.mac_address }
+        expect { post :card_number, params: rfid_params, :format => :json }.to change(Rfid, :count).by(0)
+        actual = JSON.parse(response.body, symbolize_names: true)
+        expected = { success: 'RFID sign out' }
+        expect(actual).to eq(expected)
+      end
+
+      it 'should not create rfid and should sign in RFID at one pi and sign out at other pi' do
+        sign_in_pi = create(:pi_reader)
+        sign_out_pi = create(:pi_reader, space_id: sign_in_pi.space_id)
+        attributes = create(:rfid, mac_address: sign_in_pi.pi_mac_address)
+        sign_in_rfid_params = { rfid: attributes.card_number, mac_address: sign_in_pi.pi_mac_address }
+        expect { post :card_number, params: sign_in_rfid_params, :format => :json }.to change(Rfid, :count).by(0)
+        actual = JSON.parse(response.body, symbolize_names: true)
+        sign_out_rfid_params = { rfid: attributes.card_number, mac_address: sign_out_pi.pi_mac_address }
+        expected = { success: 'RFID sign in' }
+        expect(actual).to eq(expected)
+        expect { post :card_number, params: sign_out_rfid_params, :format => :json }.to change(Rfid, :count).by(0)
+        actual = JSON.parse(response.body, symbolize_names: true)
+        expected = { success: 'RFID sign out' }
+        expect(actual).to eq(expected)
+      end
+
+      # #Temporary RFID Multi-box
```

Comment:
Is there a reason why there is a commented test?

Reply:
I wasn't sure what the logic was for temporary RFID, whether they would need to be tested in a multi box setup or not